File: ticketing-bluej/Tecnico.java
import java.util.ArrayList;
import java.util.List;

/**
 * Usuario tecnico que puede recibir tickets asignados segun su especialidad.
 */
public class Tecnico extends Usuario {
    private final String especialidad;
    private final int nivelExperiencia;
    private final List<Ticket> ticketsAsignados;

    public Tecnico(String id, String nombre, String email, String especialidad, int nivelExperiencia) {
        super(id, nombre, email);
        this.especialidad = especialidad;
        this.nivelExperiencia = nivelExperiencia;
        this.ticketsAsignados = new ArrayList<Ticket>();
    }

    public String getEspecialidad() {
        return especialidad;
    }

    public int getNivelExperiencia() {
        return nivelExperiencia;
    }

    public List<Ticket> getTicketsAsignados() {
        return ticketsAsignados;
    }

    @Override
    public void mostrarInformacion() {
        System.out.println("[TECNICO] " + getNombre() + " (ID: " + getId() + ")");
        System.out.println("  Email: " + getEmail());
        System.out.println("  Especialidad: " + especialidad);
        System.out.println("  Nivel de experiencia: " + nivelExperiencia);
        System.out.println("  Carga actual: " + obtenerCargaActual() + " ticket(s) activos");
    }

    /**
     * Agrega un ticket a la lista de tickets asignados a este tecnico.
     */
    public void asignarTicket(Ticket ticket) {
        ticketsAsignados.add(ticket);
    }

    /**
     * Cuenta los tickets asignados que aun no estan cerrados (carga de trabajo activa).
     */
    public int obtenerCargaActual() {
        int carga = 0;

        for (Ticket ticket : ticketsAsignados) {
            if (!ticket.getEstado().equals("Cerrado")) {
                carga++;
            }
        }

        return carga;
    }
}

File: logica/Tecnico.cs
namespace TicketsSoporte.logica
{
    public class Tecnico : Usuario
    {
        public string strEspecialidad;
        public int intNivelExperiencia;
        public List<Ticket> lstTicketsAsignados;

        public Tecnico(string strId, string strNombre, string strEmail, string strEspecialidad, int intNivelExperiencia)
            : base(strId, strNombre, strEmail)
        {
            this.strEspecialidad = strEspecialidad;
            this.intNivelExperiencia = intNivelExperiencia;
            lstTicketsAsignados = new List<Ticket>();
        }

        public override void mostrarInformacion()
        {
            Console.ForegroundColor = ConsoleColor.Blue;
            Console.WriteLine($"[TECNICO] {strNombre} (ID: {strId})");
            Console.ResetColor();
            Console.WriteLine($"  Email: {strEmail}");
            Console.WriteLine($"  Especialidad: {strEspecialidad}");
            Console.WriteLine($"  Nivel de experiencia: {intNivelExperiencia}");
            Console.WriteLine($"  Carga actual: {obtenerCargaActual()} ticket(s) activos");
        }

        public void asignarTicket(Ticket objTicket)
        {
            lstTicketsAsignados.Add(objTicket);
        }

        public int obtenerCargaActual()
        {
            int intCarga = 0;

            foreach (Ticket objTicket in lstTicketsAsignados)
            {
                if (objTicket.strEstado != "Cerrado")
                {
                    intCarga++;
                }
            }

            return intCarga;
        }
    }
}
